src/components/network/line-layer.js

```diff
@@ -544,267 +1027,36 @@ class LineLayer extends CompositeLayer {
-                    ),
-                    updateTriggers: {
-                        getLineParallelIndex: [this.props.lineParallelPath],
-                        getSourcePosition: [this.props.lineFullPath],
-                        getTargetPosition: [this.props.lineFullPath],
-                        getLineAngle: [this.props.lineFullPath],
-                        getProximityFactor: [this.props.lineFullPath],
-                        getColor: [
-                            this.props.disconnectedLineColor,
-                            this.props.lineFlowColorMode,
-                            this.props.lineFlowAlertThreshold,
-                            this.props.updatedLines,
-                        ],
-                    },
-                })
-            );
-            layers.push(endFork);
-
-            // lines active power
-            const lineActivePowerLabelsLayer = new TextLayer(
-                this.getSubLayerProps({
-                    id: 'ActivePower' + compositeData.nominalVoltage,
-                    data: compositeData.activePower,
-                    getText: (activePower) =>
-                        activePower.p !== undefined
-                            ? Math.round(activePower.p).toString()
-                            : '',
-                    getPosition: (activePower) => activePower.printPosition,
-                    getColor: this.props.labelColor,
-                    fontFamily: 'Roboto',
-                    getSize: this.props.labelSize,
-                    getAngle: 0,
-                    getPixelOffset: (activePower) => activePower.offset,
-                    getTextAnchor: 'middle',
-                    visible:
-                        this.props.filteredNominalVoltages.includes(
-                            compositeData.nominalVoltage
-                        ) && this.props.labelsVisible,
-                    updateTriggers: {
-                        getPosition: [
-                            this.props.lineFullPath,
-                            this.props.lineParallelPath,
-                        ],
-                        getPixelOffset: [this.props.lineFullPath],
-                    },
-                })
-            );
-            layers.push(lineActivePowerLabelsLayer);
-        });
+    genLineKey(line) {
+        return line.voltageLevelId1 > line.voltageLevelId2
+            ? line.voltageLevelId1 + '##' + line.voltageLevelId2
+            : line.voltageLevelId2 + '##' + line.voltageLevelId1;
+    }
 
-        return layers;
+    /*
+     *  function used to call onNext in the same 'frame' on the next one if we're short on time
+     *   timestamp : when the 'frame' started
+     *   onNext : callback to call, if undefined does nothing
+     * */
+    scheduleNext(timestamp, onNext) {
+        if (onNext !== undefined)
+            window.setTimeout(() => onNext(performance.now()), 0);
+    }
+
+    workArray(array, cb, currentIndex) {
```

Comment:
I don't think workArray in its current form does what you think it does. To me, it does respect tsThreshold because after the while loop, it just reexcutes itself with a new start.. but it doesn't yield execution at all.

What was workArray supposed to do ?

Reply:
missing a scheduleNext... It was supposed to cut long loops between frames.